python/dgl/nn/mxnet/softmax.py

```diff
@@ -51,8 +51,8 @@ def forward(self, score):
             g.update_all(fn.copy_e('s', 'm'), fn.max('m', 'smax'))
             g.apply_edges(fn.e_sub_v('s', 'smax', 'out'))
             g.edata['out'] = g.edata['out'].exp()
-            g.update_all(fn.copy_e('out', 'm'), fn.sum('m', 'out_sum'))
-            g.apply_edges(fn.e_div_v('out', 'out_sum', 'out'))
+            if g.number_of_edges() > 0:
+                g.apply_edges(fn.e_div_v('out', 'out_sum', 'out'))
```

Comment:
Did you accidentally delete the `update_all` call?

Reply:
Also I feel it's better to add the `g.number_of_edges() > 0` condition directly to `apply_edges`.